File: src/lib/tauri-api/coordination-artifacts.ts
import { invokeCommand } from './client';
import type {
  CoordinationArtifact,
  CreateCoordinationArtifactInput,
  UpdateCoordinationArtifactInput,
  UpdateCoordinationArtifactStatusInput,
} from '../../types/coordination-artifact';

export function listCoordinationArtifacts(workspaceId: string): Promise<CoordinationArtifact[]> {
  return invokeCommand<CoordinationArtifact[]>('list_coordination_artifacts', { workspaceId });
}

export function createCoordinationArtifact(
  input: CreateCoordinationArtifactInput,
): Promise<CoordinationArtifact> {
  return invokeCommand<CoordinationArtifact>('create_coordination_artifact', { input });
}

export function updateCoordinationArtifact(
  input: UpdateCoordinationArtifactInput,
): Promise<CoordinationArtifact> {
  return invokeCommand<CoordinationArtifact>('update_coordination_artifact', { input });
}

export function updateCoordinationArtifactStatus(
  input: UpdateCoordinationArtifactStatusInput,
): Promise<CoordinationArtifact> {
  return invokeCommand<CoordinationArtifact>('update_coordination_artifact_status', { input });
}

export function deleteCoordinationArtifact(artifactId: string): Promise<void> {
  return invokeCommand<void>('delete_coordination_artifact', { artifactId });
}

File: src-tauri/src/commands/coordination_artifacts.rs
use tauri::State;

use crate::models::{
    CoordinationArtifact, CreateCoordinationArtifactInput, UpdateCoordinationArtifactInput,
    UpdateCoordinationArtifactStatusInput,
};
use crate::services::coordination_artifact_service;
use crate::state::AppState;

#[tauri::command]
pub fn list_coordination_artifacts(
    state: State<'_, AppState>,
    workspace_id: String,
) -> Result<Vec<CoordinationArtifact>, String> {
    coordination_artifact_service::list_coordination_artifacts(&state, &workspace_id)
}

#[tauri::command]
pub fn create_coordination_artifact(
    state: State<'_, AppState>,
    input: CreateCoordinationArtifactInput,
) -> Result<CoordinationArtifact, String> {
    coordination_artifact_service::create_coordination_artifact(&state, input)
}

#[tauri::command]
pub fn update_coordination_artifact(
    state: State<'_, AppState>,
    input: UpdateCoordinationArtifactInput,
) -> Result<CoordinationArtifact, String> {
    coordination_artifact_service::update_coordination_artifact(&state, input)
}

#[tauri::command]
pub fn delete_coordination_artifact(
    state: State<'_, AppState>,
    artifact_id: String,
) -> Result<(), String> {
    coordination_artifact_service::delete_coordination_artifact(&state, &artifact_id)
}

#[tauri::command]
pub fn update_coordination_artifact_status(
    state: State<'_, AppState>,
    input: UpdateCoordinationArtifactStatusInput,
) -> Result<CoordinationArtifact, String> {
    coordination_artifact_service::update_coordination_artifact_status(&state, input)
}
